CRM/Contact/BAO/Contact.php

```diff
@@ -2629,6 +2629,79 @@ public static function getCountComponent($component, $contactId, $tableName = NU
     }
   }
 
+  /**
+   * Check whether greetings for this contact contain tokens for the given
+   * custom field.
+   *
+   * @param object $contact Contact object.
+   * @param int $customFieldId System ID of the custom field.
+   * @return bool
+   */
+  public static function checkGreetingsUseCustomFieldToken($contact, $customFieldId) {
+    $greetingStrings = self::getGreetingStrings($contact);
+    foreach ($greetingStrings as $greetingString) {
+      if (strpos($greetingString, ".custom_{$customFieldId}}") !== FALSE) {
+        return TRUE;
+      }
+    }
+    return FALSE;
+  }
+
+  /**
+   * Get an array of template strings for greeting fields.
+   *
+   * @param object $contact Contact object.
+   *
+   * @return Array of greeting strings, with keys: 'email', 'postal', 'addressee'.
+   */
+  public static function getGreetingStrings($contact) {
```

Comment:
When I looked at some of the other problems around this function I added a function (not merged) ensureGreetingParamsAreSet

The point was to have one function to ensure that those vars are set because currently it's done in multiple places. Once we get past the test issue you have we should see if we can reconcile the 2 functions a bit.

https://github.com/civicrm/civicrm-core/commit/32f50d861ca155225f9edf2c0b73120948c76ccb